python/cuml/cuml/internals/device_type.py

```diff
@@ -33,9 +33,9 @@ def from_str(cls, device_type):
 
     def is_compatible(self, mem_type: MemoryType) -> bool:
         if self is DeviceType.device:
-            return mem_type.is_device_accessible
+            return mem_type is MemoryType.device
         else:
-            return mem_type.is_host_accessible
+            return mem_type is MemoryType.host
```

Comment:
That's not the same thing if we consider managed memory.

Reply:
I agree, but `MemoryType.managed` cannot end up here (or anywhere) given our current code. This is effectively the same thing.

Reply:
Fine to simplify here for now based on the outcome of [this discussion](https://github.com/rapidsai/cuml/pull/7501#discussion_r2539698431).